source/concurrent/SynchronizedResource.ooc

```diff
@@ -0,0 +1,81 @@
+/* This file is part of magic-sdk, an sdk for the open source programming language magic.
+ *
+ * Copyright (C) 2016 magic-lang
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+use base
+use concurrent
+import threading/Thread
+import threading/Mutex
+import structs/HashMap
+
+SynchronizedResource: abstract class {
+	_threadAffinity := 0L
+	_recycle := true
+	init: func { this _threadAffinity = Thread currentThreadId() }
+	checkThreadAffinity: func -> Bool {
+		Thread equals(this _threadAffinity, Thread currentThreadId())
+	}
+}
+
+SynchronizedResourceRecycler: class {
+	_mutex := Mutex new()
+	_resources := HashMap<Long, VectorList<SynchronizedResource>> new()
```

Comment:
Because the key is not of type `Int` (or rather doesn't have its size) the hashmap will not use the int hash but rather the more complicated generic `murmurHash`. I don't really know too much about hashing functions to know whether this is good or bad, but it certainly is less efficient.